### PR TITLE
lldb: fix on Darwin

### PR DESCRIPTION
**Pull request by matthewbauer**
This gives a few fixes for LLDB on Darwin.

Note: you will have to do code signing for it to work outside of root!

###### Motivation for this change

LLDB was not supporting Darwin for some reason. This should make things work fine. Codesigning is of course not doable currently.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
Does this work when combined with #26099? Would expect so but just wanted to check :).

Regardless, LGTM.

**Comment by matthewbauer** [comment]
i'll look into it and maybe even rebase onto to avoid merge conflicts

**Comment by matthewbauer** [comment]
@dtzWill Seems to work fine! I've rebased on your PR so make sure to wait until it merges before you merge this PR.

**Comment by LnL7** [comment]
@matthewbauer can you rebase again

**Comment by matthewbauer** [comment]
Got it

**Review by LnL7 — APPROVED** [review]

**Pull request merged**
